packages/firestore/src/remote/watch_change.ts

```diff
@@ -389,33 +438,40 @@ export class WatchChangeAggregator {
 
     let resolvedLimboDocuments = documentKeySet();
 
-    this.documentTargetMapping.forEach((key, targets) => {
-      let isLimboTarget = true;
+    // We extract the set of limbo-only document updates as the GC logic
+    // special-cases documents that do not appear in the query cache.
+    //
+    // TODO(gsoltis): Expand on this comment once GC is available in the JS
+    // client.
```

Comment:
Thank you for the comment, but I'm still struggling to make sense of this...

1) It seems like your PR changes the logic significantly.  It used to be that when we got a RemoteEvent for a target used for a limbo lookup, then we added the limbo document to this list.  But now the code seems to be going out of the way to make sure the document isn't _also_ in a non-limbo target? Is this change intentional?
2) Since this is only based off pendingDocumentTargetMapping which gets cleared between every remote event, I'm not sure it actually works in a meaningful way.  The document could belong to other non-limbo targets, but those targets didn't happen to get any mapping updates since the last snapshot, right?

If we can't resolve this without Greg, perhaps expand the TODO to flag these potential issues?